src/ft60x.rs

```diff
@@ -7,151 +7,118 @@ use crate::ft60x_config::FT60xConfig;
 use crate::ringbuf::{RingBuf, RingBufConsumer};
 use crate::Result;
 use failure::format_err;
+use lazy_static::lazy_static;
 
 pub const DEFAULT_PID: u16 = 0x601f;
 pub const DEFAULT_VID: u16 = 0x0403;
 
-pub struct FT60x {
-    vid: u16,
-    pid: u16,
-    context: Context,
+lazy_static! {
+    static ref CONTEXT: Context = Context::new().unwrap();
 }
 
+pub struct FT60x {
+    device: DeviceHandle<'static>,
+    streaming_mode: bool,
+}
 impl FT60x {
-    pub fn new(vid: u16, pid: u16) -> Result<FT60x> {
-        let context = Context::new()?;
-
-        Ok(FT60x { vid, pid, context })
-    }
-
-    fn with_device_helper<F: FnMut(&mut DeviceHandle) -> Result<T>, T>(
-        context: &Context,
-        vid: u16,
-        pid: u16,
-        mut func: F,
-    ) -> Result<T> {
-        let mut device = context.open_device_with_vid_pid(vid, pid).ok_or_else(|| {
-            format_err!("No device with VID {:#x} and PID {:#x} was found", vid, pid)
-        })?;
-
-        Ok(func(&mut device)?)
-    }
-
-    fn with_device<F: FnMut(&mut DeviceHandle) -> Result<T>, T>(&self, func: F) -> Result<T> {
-        Self::with_device_helper(&self.context, self.vid, self.pid, func)
+    pub fn new(vid: u16, pid: u16) -> Result<Self> {
+        let device =
+            CONTEXT
+                .open_device_with_vid_pid(vid, pid)
+                .ok_or_else(|| format_err!("No device with VID {:#x} and PID {:#x} was found", vid, pid))?;
+        Ok(Self { device, streaming_mode: false })
     }
 
-    pub fn get_config(&self) -> Result<FT60xConfig> {
+    pub fn get_config(&mut self) -> Result<FT60xConfig> {
```

Comment:
I am not sure how to handle `mut` and non `mut` parts for the public API, naivly speaking as a external user I would assume something that only gets something should be non `mut`?

Reply:
right that is a bit unfortunate but I think it is okay as is for now